edk2toolext/edk2_invocable.py

```diff
@@ -209,13 +209,14 @@ def collect_python_pip_info(cls: "Edk2Invocable") -> None:
             pip_packages = []
         # go through all installed pip versions
         for package in pip_packages:
-            try:
-                version = package.version
-                name = package.metadata["Name"] if "Name" in package.metadata else package.metadata.get("name", None)
-                if not name:
-                    name = package.metadata.get("Summary", "unknown")
-            except Exception:
+            name = package.metadata.get("Name", None) or package.metadata.get("Summary", None)
+            if name is None:
                 continue
+
+            # Always grab the version of the first instance of the package on the PATH. pipenv inserts its
+            # path first, so we can assume that the first instance is the one actually used.
+            version = importlib.metadata.version(name)
+
             logging.info("{0} version: {1}".format(name, version))
```

Comment:
Nitpick: This will still print duplicate pip modules in the build output. This could be filtered to prevent that.
Example - 
BUILDLOG.txt
278: INFO - typing_extensions version: 4.14.0
293: INFO - typing_extensions version: 4.14.0

This should not be considered a blocking issue.